File: selenium/src/test/java/com/ytfc/blog/forms/LoginForm.java
package com.ytfc.blog.forms;

import java.util.HashMap;

import com.ytfc.blog.forms.login.LoginSubmit;
import com.ytfc.blog.forms.login.NameField;
import com.ytfc.blog.forms.login.PasswordField;

import org.openqa.selenium.WebDriver;

public class LoginForm extends FormObject {

    public LoginForm(WebDriver driver) {
        super(driver);

        fields = new HashMap<String, FormField>();

        fields.put("name", new NameField(driver));
        fields.put("password", new PasswordField(driver));

        button = new LoginSubmit(driver);
    }


}

File: cypress/forms/login-form.js
import { FormBase   } from './form-base'
import { FormField  } from './form-field'
import { FormButton } from './form-button'

export class LoginForm extends FormBase {
    constructor() {
        super();
        this._fields.name      = new FormField('text',     'user-login-name')
        this._fields.password  = new FormField('password', 'user-login-password')

        this._buttons.submit   = new FormButton('login-button')
    }
}
